src/transformers/tokenization_utils.py

```diff
@@ -1410,3 +1410,147 @@ def clean_up_tokenization(out_string):
             .replace(" 're", "'re")
         )
         return out_string
+
+
+class FastPreTrainedTokenizer(PreTrainedTokenizer):
+    def __init__(self, **kwargs):
+        super(FastPreTrainedTokenizer, self).__init__(**kwargs)
+
+    @property
+    def tokenizer(self):
+        if self._tokenizer is None:
+            raise NotImplementedError
+        return self._tokenizer
+
+    @property
+    def decoder(self):
+        if self._decoder is None:
+            raise NotImplementedError
+        return self._decoder
+
+    @property
+    def vocab_size(self):
+        return self.tokenizer.get_vocab_size(False)
+
+    def __len__(self):
+        return self.tokenizer.get_vocab_size(True)
+
+    def _update_special_tokens(self):
+        self.tokenizer.add_special_tokens(self.all_special_tokens)
+
+    @staticmethod
+    def _convert_encoding(
+        encoding,
+        return_tensors=None,
+        return_token_type_ids=True,
+        return_attention_mask=True,
+        return_overflowing_tokens=False,
+        return_special_tokens_mask=False,
+    ):
+        encoding_dict = {
+            "input_ids": encoding.ids,
+        }
+        if return_token_type_ids:
+            encoding_dict["token_type_ids"] = encoding.type_ids
+        if return_attention_mask:
+            encoding_dict["attention_mask"] = encoding.attention_mask
+        if return_overflowing_tokens:
+            overflowing = encoding.overflowing
+            encoding_dict["overflowing_tokens"] = overflowing.ids if overflowing is not None else []
+        if return_special_tokens_mask:
+            encoding_dict["special_tokens_mask"] = encoding.special_tokens_mask
```

Comment:
Does `encoding` contain tokens to char/byte mappings as well?

We could take the occasion of the `tokenizers` project to add this features as well.
This is an important feature. Current status is that we are waiting to see how `tokenizers` will handle it to add matching python implementations for the non-fast tokenizers.

Reply:
Not at the moment. I have all the structure in place to handle this, but it doesn't always work for now. The GPT2Tokenizer is handled out of the box since there's no magic done, but I can't say the same for BertTokenizer. I'll need to handle the Normalization and PreTokenization steps properly first before we can expose the offsets to the user.